lib/inferno/dsl/fhir_evaluator/evaluator_util.rb

```diff
@@ -0,0 +1,81 @@
+module Util
+  # General util methods
+  def self.get_meta_profile(resource)
+    resource&.meta&.profile || []
+  end
+
+  # reference resolution methods
+  def self.extract_ids_references(resources)
+    extractor = ReferenceExtractor.new
+    extractor.extract_ids_references(resources)
+  end
+
+  class ReferenceExtractor
+    def initialize
+      @resource_type_ids = Hash.new { |h, k| h[k] = [] }
+      @resource_ids = Set.new
+      @references = Hash.new { |h, k| h[k] = [] }
+    end
+
+    def extract_ids_references(resources)
```

Comment:
This method needs a better name. It's not at all clear to me what this does based on its name.